app/javascript/packs/components/Punches.js

```diff
@@ -1,13 +1,28 @@
-import React from 'react';
-import Punch from './Punch';
+import React from "react";
+import Punch from "./Punch";
+import { isEmpty } from "lodash";
 
 const Punches = ({ sheet }) => {
```

Comment:
TIP: I think that you could improve a little bit the way how this component is rendered. Like this:
```jsx
const Punches = ({ sheet }) => {
  if (isEmpty(sheet)) return null;

  const workedHours = () => {
    const totalDelta = sheet.reduce(
      (acc, current) => (acc += current.delta),
      0
    );

    return `Total: ${totalDelta}h`;
  };

  return (
    <ul className="punches">
      {sheet.map((punch, i) => (
        <Punch key={i} punch={punch} />
      ))}
      <li>{workedHours(sheet)}</li>
    </ul>
  );
};

export default Punches;
```